Cilsil/Services/CfgParserService.cs

```diff
@@ -88,50 +103,89 @@ private void ComputeMethodCfg(MethodDefinition method)
 
             // True if the translation terminates early, false otherwise.
             var translationUnfinished = false;
-            if (!method.IsAbstract && methodBody.Instructions.Count > 0)
+            foreach (var exceptionHandler in methodBody.ExceptionHandlers)
             {
-                programState.PushInstruction(methodBody.Instructions.First());
-                do
+                var exceptionHandlingBlock = exceptionHandler;
+                var exceptionHandlingBlockStartOffset = -1;
+                var exceptionHandlingBlockEndOffset = -1;
+                TypeReference catchType = null;
+                try
                 {
-                    var nextInstruction = programState.PopInstruction();
-                    // Checks if there is a node for the offset that we can reuse.
-                    (var nodeAtOffset, var excessiveVisits) =
-                        programState.GetOffsetNode(nextInstruction.Offset);
-                    if (nodeAtOffset != null)
-                    {
-                        programState.PreviousNode.Successors.Add(nodeAtOffset);
-                    }
-                    else if (excessiveVisits)
+                    switch (exceptionHandlingBlock.HandlerType)
                     {
-                        TimeoutMethodCount++;
-                        Log.WriteError("Translation timeout.");
-                        Log.RecordUnfinishedMethod(programState.Method.GetCompatibleFullName(),
-                                                   nextInstruction.RemainingInstructionCount());
-                        translationUnfinished = true;
-                        break;
+                        case ExceptionHandlerType.Catch:
+                            catchType = exceptionHandlingBlock.CatchType;
+                            exceptionHandlingBlockStartOffset = exceptionHandlingBlock.HandlerStart.Offset;
+                            exceptionHandlingBlockEndOffset = exceptionHandlingBlock.HandlerEnd.Offset;
+                            break;
+
+                        case ExceptionHandlerType.Finally:
```

Comment:
Why aren't we recording catchType in this case?

Reply:
Finally block doesn't have any catch type, for it does not for exception handling purpose. As you can see, catch type does not exist in the CIL as well.
![image](https://user-images.githubusercontent.com/63893533/122620882-f44f6a00-d048-11eb-8e33-07fca31ac075.png)